unblob/pool.py

```diff
@@ -72,11 +83,30 @@ def __init__(
             for _ in range(process_num)
         ]
         self._tid = threading.get_native_id()
+        self._signals_fired = []
+
+    def _handle_signal(self, signum: int, frame):
+        logger.debug(f"Signal ({signal.Signals(signum).name}) received")
```

Comment:
Based on our investigation of the possible deadlock introduced by the process pool approach I am a bit hesitant about this approach: If there is a mutex involved for printing log messages, this log line here can introduce a deadlock. It is generally not safe to cause any side-effect from a signal handler because of these issues creeping in.

P.s.
This also involves somewhat true for raising an exception as well, because we don't know if we are in a `try: ... except: pass` block for example when the signal arrives. (I guess this is true for KeyboardInterrupt as well...).

Reply:
Maybe we can talk about this more, but at his point, I don't understand how do we want to do it. Of course we don't have to log if that makes things harder, but we also can't raise the exception here? The whole concept if based on raising the custom exception in the signal handler. If that introduces this level of uncertainty, we may to rethink the idea?

Reply:
The exception raising can remain, it is just a remark about its limitation (which applies to the out of the box ctrl c handling as well)